src/actors/client/ActorClient/ActorClientGRPC.ts

```diff
@@ -0,0 +1,278 @@
+          return reject(err);
+        }
+
+        // https://docs.dapr.io/reference/api/actors_api/#http-response-codes-3
+        return resolve();
+      });
+    });
+  }
+
+  // @todo: not implemented, cannot find the gRPC bindings
+  // async reminderGet(actorType: string, actorId: string, name: string): Promise<void> {
+  //     const msgService = new RegisterActorReminderRequest();
+  //     msgService.setActorType(actorType);
+  //     msgService.setActorId(actorId);
+  //     msgService.setName(name);
+
+  //     if (reminder.data) {
+  //         msgService.setData(Buffer.from(reminder.data, "utf-8"))
+  //     }
+
+  //     if (reminder.period) {
+  //         msgService.setPeriod(reminder.period);
+  //     }
+
+  //     if (reminder.dueTime) {
+  //         msgService.setDueTime(reminder.dueTime);
+  //     }
+
+  //     return new Promise(async (resolve, reject) => {
+  //         const client = await GRPCClientSingleton.getClient();
+  //         client.registerActorReminder(msgService, (err, res) => {
+  //             if (err) {
+  //                 return reject(err);
+  //             }
+
+  //             // https://docs.dapr.io/reference/api/actors_api/#http-response-codes-3
+  //             return resolve();
+  //         });
+  //     });
+  // }
+
+  async reminderDelete(actorType: string, actorId: string, name: string): Promise<void> {
+    const msgService = new UnregisterActorReminderRequest();
+    msgService.setActorType(actorType);
+    msgService.setActorId(actorId);
+    msgService.setName(name);
+
+    return new Promise((resolve, reject) => {
+      const client = this.client.getClient();
+      client.unregisterActorReminder(msgService, (err, res) => {
+        if (err) {
+          return reject(err);
+        }
+
+        // https://docs.dapr.io/reference/api/actors_api/#delete-actor-reminder
+        return resolve();
+      });
+    });
+  }
+
+  async timerCreate(actorType: string, actorId: string, name: string, timer: ActorTimerType): Promise<void> {
+    const msgService = new RegisterActorTimerRequest();
+    msgService.setActorType(actorType);
+    msgService.setActorId(actorId);
+    msgService.setName(name);
+
+    if (timer.callback) {
+      msgService.setCallback(timer.callback);
+    }
+
+    if (timer.data) {
+      msgService.setData(Buffer.from(timer.data, "utf-8"))
+    }
+
+    if (timer.period) {
+      msgService.setPeriod(timer.period.toString());
+    }
+
+    if (timer.dueTime) {
+      msgService.setDueTime(timer.dueTime.toString());
+    }
+
+    return new Promise((resolve, reject) => {
+      const client = this.client.getClient();
+      client.registerActorTimer(msgService, (err, res) => {
+        if (err) {
+          return reject(err);
+        }
+
+        // https://docs.dapr.io/reference/api/actors_api/#http-response-codes-3
+        return resolve();
+      });
+    });
+  }
+
+  async timerDelete(actorType: string, actorId: string, name: string): Promise<void> {
```

Comment:
I wonder if these methods should be called deleteTimer, createTimer, etc. We went through an exercise to normalize naming among the SDKs a while ago and only accept variations that are idiomatic to each language. See the example in Java SDK:

https://github.com/dapr/java-sdk/blob/master/sdk-actors/src/main/java/io/dapr/actors/runtime/AbstractActor.java

Reply:
Ok will change